### PR TITLE
Add class for handling restart initialization in BlackoilWellModel

### PR DESCRIPTION
**Pull request by akva2**
More well code splitting.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — COMMENTED** [review]
I like it and it helps decoupling the well code which is definitely laudable.  Just one minor naming issue and then I think this is ready to go in.

File: opm/simulators/wells/BlackoilWellModelRestart.hpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                             const std::vector<data::Rates::opt>& phs,
+                             const data::Well&                    rst_well,
+                             const std::vector<PerforationData>&  old_perf_data,
+                             SingleWellState&                     ws) const;
+
+    void loadRestartGroupData(const std::string&     group,
+                              const data::GroupData& value,
+                              GroupState& grpState) const;
+
+    const BlackoilWellModelGeneric& well_;
```
[the remark this answers is not in the document]
Comment (akva2) [reply opm/simulators/wells/BlackoilWellModelRestart.hpp]: yeah, i've been working on the wellif earlier so i was on a rail. changed to wellModel_

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**